packages/core/src/wallet/IndyWallet.ts

```diff
@@ -171,31 +179,31 @@ export class IndyWallet implements Wallet {
    * @throws {WalletError} if another error occurs
    */
   public async delete(): Promise<void> {
-    const walletInfo = this.openWalletInfo
-
-    if (!this.isInitialized || !walletInfo) {
+    if (!this.walletConfig) {
       throw new WalletError(
-        'Can not delete wallet that is not initialized. Make sure to call initialize before deleting the wallet'
+        'Can not delete wallet that does not have wallet config set. Make sure to call create wallet before deleting the wallet'
       )
     }
 
-    this.logger.info(`Deleting wallet '${walletInfo.walletConfig.id}'`)
+    this.logger.info(`Deleting wallet '${this.walletConfig.id}'`)
 
-    await this.close()
+    if (this.walletHandle) {
```

Comment:
Good point. But this one is tricky.

As I'm thinking about it now, I don't think it's a good idea to just silently do nothing if the developer calls `close` and `walletHandle` is not set, because it can lead to situations when the developer accidentally sets `agent.wallet.walletHandle = undefined` but wallet will be still opened. It should actually throw an error when you try to call `close` directly and `walletHandle` is not set. TypeScript doesn't allow me to call only `await this.indy.closeWallet(this.walletHandle)`, it says:

```
Argument of type 'number | undefined' is not assignable to parameter of type 'number'.
  Type 'undefined' is not assignable to type 'number'.ts(2345)
```

Therefore, I should validate `this.walletHandle` and eventually throw an error before calling `await this.indy.closeWallet(this.walletHandle)`.

However, if I'm deleting a wallet that has not been opened yet, I don't want to call `close` on it because that could throw an error I described above. So, although I don't like this double-check (each time for a different reason), I think it's a more robust solution.